supervisor/dbus/network/settings.py

```diff
@@ -29,11 +31,16 @@ async def connect(self) -> None:
             )
 
     @dbus_connected
-    def add_connection(self, settings: Any) -> Awaitable[Any]:
+    async def add_connection(self, settings: Any) -> NetworkSetting:
         """Add new connection."""
-        return self.dbus.Settings.AddConnection(("a{sa{sv}}", settings))
+        obj_con_setting = (
+            await self.dbus.Settings.AddConnection(("a{sa{sv}}", settings))
+        )[0]
+        con_setting = NetworkSetting(obj_con_setting)
+        await con_setting.connect()
+        return con_setting
```

Comment:
This seemed more consistent with how we handle settings elsewhere. However the only caller of this method today is here:
https://github.com/home-assistant/supervisor/blob/01e24a3e74c54c4b93e4c87196d1a5ebf8d0fb42/supervisor/host/network.py#L221

Notice it does not use the output. So this does create a new situation where we added the setting without error but got an error on `connect`. We wouldn't have encounted this before. I'm not sure how this could possibly occur but if somehow we can't connect to the new settings that feels like a problem we should report.